qiskit/primitives/backend_estimator_v2.py

```diff
@@ -0,0 +1,262 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Estimator V2 implementation for an arbitrary BackendV2 object."""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from collections.abc import Iterable
+
+import numpy as np
+
+from qiskit.circuit import ClassicalRegister, QuantumCircuit, QuantumRegister
+from qiskit.exceptions import QiskitError
+from qiskit.providers import BackendV2
+from qiskit.quantum_info import Pauli, PauliList
+from qiskit.transpiler import PassManager, PassManagerConfig
+from qiskit.transpiler.passes import Optimize1qGatesDecomposition
+
+from .backend_estimator import _pauli_expval_with_variance, _prepare_counts, _run_circuits
+from .base import BaseEstimatorV2
+from .containers import EstimatorPubLike, PrimitiveResult, PubResult
+from .containers.bindings_array import BindingsArray
+from .containers.estimator_pub import EstimatorPub
+from .primitive_job import PrimitiveJob
+
+
+class BackendEstimatorV2(BaseEstimatorV2):
+    """Evaluates expectation value using Pauli rotation gates.
+
+    The :class:`~.BackendEstimator` class is a generic implementation of the
+    :class:`~.BaseEstimator` interface that is used to wrap a :class:`~.BackendV2`
+    (or :class:`~.BackendV1`) object in the :class:`~.BaseEstimator` API. It
+    facilitates using backends that do not provide a native
+    :class:`~.BaseEstimator` implementation in places that work with
+    :class:`~.BaseEstimator`. However,
+    if you're using a provider that has a native implementation of
+    :class:`~.BaseEstimator`, it is a better choice to leverage that native
+    implementation as it will likely include additional optimizations and be
+    a more efficient implementation. The generic nature of this class
+    precludes doing any provider- or backend-specific optimizations.
+
+    Implementation of :class:`BaseEstimatorV2` using a backend.
+
+    This class provides a EstimatorV2 interface from any :class:`~.BackendV2` backend
+    and doesn't do any measurement mitigation, it just computes the expectation values.
+    At present, this implementation is only compatible with Pauli-based observables.
+
+    Each tuple of ``(circuit, observables, <optional> parameter values, <optional> precision)``,
+    called an estimator primitive unified bloc (PUB), produces its own array-based result. The
+    :meth:`~.EstimatorV2.run` method can be given a sequence of pubs to run in one call.
+    """
+
+    _VARIANCE_UPPER_BOUND: float = 4.0
+
+    def __init__(
+        self,
+        *,
+        backend: BackendV2,
+        default_precision: float = 0.0,
+        abelian_grouping: bool = True,
+    ):
+        """
+        Args:
+            backend: Required: the backend to run the primitive on
+            default_precision: Default precision
+            abelian_grouping: Whether the observable should be grouped into commuting
```

Comment:
This sentence is unfinished. I'm also wondering if we can somehow change the name away from abelian, so as not to imply that XX and YY are grouped. `group_qubitwise_commuting` or something? However, not a big deal, more important is that this is mentioned in the docstring.

```suggestion
            abelian_grouping: Whether the observable should be grouped into commuting
```

Reply:
I copied it from `BackendEstimator`. But as you mentioned, it looks like unfinished. I revised it.